src/elements/content-sidebar/Sidebar.js

```diff
@@ -103,8 +107,40 @@ class Sidebar extends React.Component<Props, State> {
             this.setForcedByLocation();
             this.setState({ isDirty: true });
         }
+
+        // Active annotation id changed. If location is currently an annotation path or
+        // if location is not currently an annotation path but the active annotation id
+        // transitioned from falsy to truthy, update the location accordingly
+        if (
+            prevActiveAnnotationId !== activeAnnotationId &&
+            (isAnnotationsPath || (activeAnnotationId && !isAnnotationsPath))
```

Comment:
Could pull some of these conditions out into variables to help with readability?